src/app/app-routing.module.ts

```diff
@@ -4,16 +4,25 @@ import { RouterModule, Routes } from '@angular/router';
 import { CoursesPageComponent } from '@app/courses-page';
 import { LoginPageComponent } from '@app/login-page';
 import { NewCoursesPageComponent } from '@app/new-courses-page';
+import { ErrorPageComponent } from '@app/error-page';
+import { AuthGuard } from '@app/core';
 
 const routes: Routes = [
-  { path: 'courses', component: CoursesPageComponent },
+  { 
+    path: 'courses',
+    canActivate: [AuthGuard],
+    children: [
+    { path: '', component: CoursesPageComponent },
+    { path: ':id', component: NewCoursesPageComponent },
+    { path: 'new', component: NewCoursesPageComponent }
```

Comment:
{ path: 'new', component: NewCoursesPageComponent }
{ path: ':id', component: NewCoursesPageComponent },
 switch those config so it will work properly. `:id` will match anything, so you need to put more specific route at the top

Reply:
@SteveRusin Thanks. I have changed links order.